src/bundle/Resources/config/backup_codes.xml

```diff
@@ -6,7 +6,5 @@
 		</service>
 
 		<service id="scheb_two_factor.null_backup_code_manager" class="Scheb\TwoFactorBundle\Security\TwoFactor\Backup\NullBackupCodeManager" />
-
-		<service id="scheb_two_factor.backup_code_manager" alias="scheb_two_factor.null_backup_code_manager" />
```

Comment:
If this PR is for a next major, we should be able to define these alias as deprecated on the previous version with something like that : https://github.com/doctrine/DoctrineBundle/blob/1.12.x/DependencyInjection/DoctrineExtension.php#L95-L99
Otherwise it will be a BC break

Reply:
Indeed the alias still exist. It is created by the bundle exension (see below) https://github.com/scheb/two-factor-bundle/blob/ca10dd6809329f7c51b15bd5a74c4ea479ec6d3f/src/bundle/DependencyInjection/SchebTwoFactorExtension.php#L76

The only difference is that before the alias was `scheb_two_factor.null_backup_code_manager` and replaced if the associated configuration option is set.
Now the option is set by default in Configuration file
https://github.com/scheb/two-factor-bundle/blob/ca10dd6809329f7c51b15bd5a74c4ea479ec6d3f/src/bundle/DependencyInjection/Configuration.php#L68

Reply:
I missed it, forget my comment 😅

Reply:
No problem.
I still see a modification ofthe behaviour here. If the backup code feature is not enabled, the alias is not created. With this alias in the xml file, the alias is always set even if the disabled.